File: manual/data/python/tests/test_algorithms_binary_search.py
import unittest
from function_algorithms_binary_search import BinarySearchFunction

class TestBinarySearchFunction(unittest.TestCase):
    def setUp(self):
        self.function = BinarySearchFunction()

    def test_target_present_in_string(self):
        data = "246810"
        target = '6'
        self.assertEqual(self.function.binary_search_index(data, target), 2)

    def test_target_absent_in_string(self):
        data = "246810"
        target = '5'
        self.assertEqual(self.function.binary_search_index(data, target), -1)

    def test_empty_string(self):
        data = ""
        target = '1'
        self.assertEqual(self.function.binary_search_index(data, target), -1)

    def test_single_character_found(self):
        data = "7"
        target = '7'
        self.assertEqual(self.function.binary_search_index(data, target), 0)

    def test_single_character_not_found(self):
        data = "7"
        target = '3'
        self.assertEqual(self.function.binary_search_index(data, target), -1)

if __name__ == "__main__":
    unittest.main()

File: manual/code/test_suites/cpp/FunctionBuildTest/tests/test_algorithms_binary_search.cpp
#include <gtest/gtest.h>
#include "../src/function_algorithms_binary_search.cpp"

class TestBinarySearchFunction : public ::testing::Test {
protected:
    BinarySearchFunction function;
};

TEST_F(TestBinarySearchFunction, TestTargetPresentInString) {
    std::string data = "246810";
    char target = '6';
    EXPECT_EQ(function.binary_search_index(data, target), 2);
}

TEST_F(TestBinarySearchFunction, TestTargetAbsentInString) {
    std::string data = "246810";
    char target = '5';
    EXPECT_EQ(function.binary_search_index(data, target), -1);
}

TEST_F(TestBinarySearchFunction, TestEmptyString) {
    std::string data = "";
    char target = '1';
    EXPECT_EQ(function.binary_search_index(data, target), -1);
}

TEST_F(TestBinarySearchFunction, TestSingleCharacterFound) {
    std::string data = "7";
    char target = '7';
    EXPECT_EQ(function.binary_search_index(data, target), 0);
}

TEST_F(TestBinarySearchFunction, TestSingleCharacterNotFound) {
    std::string data = "7";
    char target = '3';
    EXPECT_EQ(function.binary_search_index(data, target), -1);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}